### PR TITLE
Add compose dependencies

### PR DESCRIPTION
**Pull request by Skrilltrax**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [x] Dependency updates

## :scroll: Description
Adds compose dependencies to the Gradle build system.

## :bulb: Motivation and Context

Compose will soon reach 1.0 and we should start using it.

## :green_heart: How did you test it?
The project builds fine.

## :pencil: Checklist

- [x] I formatted the code `./gradlew spotlessApply`
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -14,6 +14,14 @@ plugins {
   `crowdin-plugin`
 }
 
+repositories {
+  // https://github.com/chrisbanes/tivi/blob/main/build.gradle
+  val composeSnapshot = libs.versions.composeSnapshot.get()
+  if (composeSnapshot.length > 1) {
```
Comment (msfjarvis) [inline app/build.gradle.kts]:
```suggestion
  if (composeSnapshot.isNotEmpty()) {
```

File: app/build.gradle.kts [hunk 2]
```diff
@@ -14,6 +14,14 @@ plugins {
...
+    maven("https://androidx.dev/snapshots/builds/$composeSnapshot/artifacts/repository/")
```
Comment (msfjarvis) [inline app/build.gradle.kts]: Needs a `content` block like the repositories defined in `BaseProjectConfig.kt`. Use `includeGroupByRegex("androidx\\.compose\\.*")` as the inclusion rule.

File: app/build.gradle.kts [hunk 3]
```diff
@@ -29,6 +37,7 @@ android {
   defaultConfig {
     applicationId = "dev.msfjarvis.aps"
     testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    vectorDrawables { useSupportLibrary = true }
```
Comment (msfjarvis) [inline app/build.gradle.kts]: Is this necessary?

File: app/build.gradle.kts [hunk 4]
```diff
@@ -37,6 +46,7 @@ android {
     disable("MissingTranslation", "PluralsCandidate", "ImpliedQuantity")
     // https://issuetracker.google.com/issues/187524311
     disable("DialogFragmentCallbacksDetector")
+    disable("CoroutineCreationDuringComposition")
```
Comment (msfjarvis) [inline app/build.gradle.kts:58]: If there's an issue filed on issuetracker.google.com already please add a link

File: app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package dev.msfjarvis.aps.ui.theme
+
+import androidx.compose.ui.graphics.Color
+
+val Purple200 = Color(0xFFBB86FC)
+val Purple500 = Color(0xFF6200EE)
+val Purple700 = Color(0xFF3700B3)
+val Teal200 = Color(0xFF03DAC5)
```
Comment (msfjarvis) [inline app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt]: Get rid of all these files, they're not adding any value.

File: buildSrc/build.gradle.kts [hunk 6]
```diff
@@ -44,12 +44,11 @@ dependencies {
   implementation("com.android.tools.build:gradle:4.2.1")
   implementation("com.diffplug.spotless:spotless-plugin-gradle:5.12.4")
   implementation("com.google.dagger:hilt-android-gradle-plugin:2.35.1")
-  implementation("com.squareup.okhttp3:okhttp:4.9.0")
+  implementation("com.squareup.okhttp3:okhttp:5.0.0-alpha.2")
```
Comment (msfjarvis) [inline buildSrc/build.gradle.kts]:
```suggestion
  implementation("com.squareup.okhttp3:okhttp:4.9.0")
```

Unrelated change with little value for the one HTTP call we make.

File: buildSrc/src/main/java/BaseProjectConfig.kt [hunk 7]
```diff
@@ -83,6 +83,7 @@ internal fun BaseAppModuleExtension.configureAndroidApplicationOptions(project:
   buildFeatures {
     viewBinding = true
     buildConfig = true
+    compose = true
```
Comment (msfjarvis) [inline buildSrc/src/main/java/BaseProjectConfig.kt]: If we're setting `compose` properties here then the repository configuration should be moved here too.

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -1,9 +1,11 @@
 # Centralized versions for dependencies that share versions
 [versions]
 androidx_test = "1.4.0-beta01"
+compose = "1.0.0-SNAPSHOT"
+composeSnapshot = "7395546"
```
Comment (msfjarvis) [inline gradle/libs.versions.toml:5]: Add a comment explaining where this is sourced from and what the relation to the compose version above is.

File: gradle/libs.versions.toml [hunk 9]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
 androidx-swiperefreshlayout = "androidx.swiperefreshlayout:swiperefreshlayout:1.2.0-alpha01"
 
+# Compose dependencies
+compose-animation-animation = { module = "androidx.compose.animation:animation", version.ref = "compose" }
+compose-compiler-compiler = { module = "androidx.compose.compiler:compiler", version.ref = "compose" }
+compose-foundation-foundation = { module = "androidx.compose.foundation:foundation", version.ref = "compose" }
+compose-foundation-layout = { module = "androidx.compose.foundation:foundation-layout", version.ref = "compose" }
+compose-material-iconsext = { module = "androidx.compose.material:material-icons-extended", version.ref = "compose" }
+compose-material-material = { module = "androidx.compose.material:material", version.ref = "compose" }
+compose-compiler-runtime = { module = "androidx.compose.runtime:runtime", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-runtime = { module = "androidx.compose.runtime:runtime", version.ref = "compose" }
```

File: gradle/libs.versions.toml [hunk 10]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+androidx-lifecycle-viewmodel-compose = "androidx.lifecycle:lifecycle-viewmodel-compose:1.0.0-alpha05"
 androidx-material = "com.google.android.material:material:1.4.0-beta01"
 androidx-preference = "androidx.preference:preference:1.1.1"
 androidx-recyclerview = "androidx.recyclerview:recyclerview:1.2.0"
 androidx-recyclerviewSelection = "androidx.recyclerview:recyclerview-selection:1.2.0-alpha01"
 androidx-security = "androidx.security:security-crypto:1.1.0-alpha03"
 androidx-swiperefreshlayout = "androidx.swiperefreshlayout:swiperefreshlayout:1.2.0-alpha01"
 
+# Compose dependencies
+compose-animation-animation = { module = "androidx.compose.animation:animation", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-animation = { module = "androidx.compose.animation:animation", version.ref = "compose" }
```

File: gradle/libs.versions.toml [hunk 11]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+compose-compiler-compiler = { module = "androidx.compose.compiler:compiler", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-compiler = { module = "androidx.compose.compiler:compiler", version.ref = "compose" }
```

File: gradle/libs.versions.toml [hunk 12]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+compose-foundation-foundation = { module = "androidx.compose.foundation:foundation", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-foundation = { module = "androidx.compose.foundation:foundation", version.ref = "compose" }
```

File: gradle/libs.versions.toml [hunk 13]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+compose-foundation-layout = { module = "androidx.compose.foundation:foundation-layout", version.ref = "compose" }
+compose-material-iconsext = { module = "androidx.compose.material:material-icons-extended", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]: We're never going to use this, the size penalty is exorbitant

File: gradle/libs.versions.toml [hunk 14]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+compose-material-material = { module = "androidx.compose.material:material", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-material = { module = "androidx.compose.material:material", version.ref = "compose" }
```

File: gradle/libs.versions.toml [hunk 15]
```diff
@@ -33,16 +36,33 @@ androidx-constraintlayout = "androidx.constraintlayout:constraintlayout:2.1.0-be
...
+compose-compiler-runtime = { module = "androidx.compose.runtime:runtime", version.ref = "compose" }
+compose-ui-test = { module = "androidx.compose.ui:ui-test-junit4", version.ref = "compose" }
+compose-ui-tooling = { module = "androidx.compose.ui:ui-tooling", version.ref = "compose" }
+compose-ui-ui = { module = "androidx.compose.ui:ui", version.ref = "compose" }
```
Comment (msfjarvis) [inline gradle/libs.versions.toml]:
```suggestion
compose-ui = { module = "androidx.compose.ui:ui", version.ref = "compose" }
```

**Review by msfjarvis — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 3 above)
Comment (Skrilltrax) [reply app/build.gradle.kts]: The Android Studio adds it automatically whenever you set up compose through it so I decided to keep it in.

File: app/build.gradle.kts (hunk 4 above)
Comment (Skrilltrax) [reply app/build.gradle.kts:58]: No issue filed yet.

File: app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt (hunk 5 above)
Comment (Skrilltrax) [reply app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt]: I specially added them so that we can use them as a template and modify the files to match our color scheme, shapes, etc. I'll remove it we don't want it in the app.

File: buildSrc/src/main/java/BaseProjectConfig.kt (hunk 7 above)
Comment (Skrilltrax) [reply buildSrc/src/main/java/BaseProjectConfig.kt]: The problem with moving `compose` properties here is that we won't be able to access the compose snapshot version present in the `libs.versions.toml`. I can move this to `app/build.gradle.kts` to keep all the stuff at one place. What are your thoughts on that?

File: gradle/libs.versions.toml (hunk 8 above)
Comment (Skrilltrax) [reply gradle/libs.versions.toml:5]: Sure

File: buildSrc/src/main/java/BaseProjectConfig.kt (hunk 7 above)
Comment (msfjarvis) [reply buildSrc/src/main/java/BaseProjectConfig.kt]: That works too

File: gradle/libs.versions.toml (hunk 13 above)
Comment (Skrilltrax) [reply gradle/libs.versions.toml]: That won't be an issue in release builds since R8 should remove most of the classes from there. I say let's keep the dependency there and we can decide about this later? or if you're 100% sure, I can just remove this now.

File: app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt (hunk 5 above)
Comment (msfjarvis) [reply app/src/main/java/dev/msfjarvis/aps/ui/theme/Color.kt]: We're gonna use the material theme adapter for the initial migration so this is going to stay unused for a good while, there's no point keeping templates around for that long.

File: gradle/libs.versions.toml (hunk 13 above)
Comment (msfjarvis) [reply gradle/libs.versions.toml]: It'll only serve to fragment icon resources between drawable XMLs and code, we're not using it.

**Review by msfjarvis — APPROVED** [review]
